File: OOYA_SP/C#/OPLab/Manager.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace OPLab
{
    public class Manager : Programmer
    {
        private int _subordinatesQuantity;
        private List<String> _projects;

        public Manager(String name, String surname, Position position, List<String> skills,
                         int subordinatesQuantity, List<String> projects) : base(name, surname, position, skills)
        {
            SubordinatesQuantity = subordinatesQuantity;
            Projects = projects;
        }

        public int SubordinatesQuantity { get => _subordinatesQuantity; set => _subordinatesQuantity = value; }
        public List<String> Projects { get => _projects; set => _projects = value; }

		public string GetProjectsAsString()
		{
			string result = Projects[0] + " ";
            foreach (string s in Projects)
            {
                result += s + " ";
            }
			return result;
		}

        public override void Print() //
        {
            Console.WriteLine("Manager: {0} {1}", Surname, Name);
            Console.WriteLine("Position: {0}", Position);
            Console.WriteLine("Skills: {0}", GetSkillsAsString());
            Console.WriteLine("Projects: {0}", GetProjectsAsString());
            Console.WriteLine("Subordinates quantity: {0}",SubordinatesQuantity); 
            Console.WriteLine();
        }
    }
}

File: OOYA_SP/Java/src/main/java/com/ran/splabjava/Manager.java
package com.ran.splabjava;

import java.util.List;

public class Manager extends Programmer {

    private List<String> projects;
    private int subordinatesQuantity;

    public Manager(String name, String surname, Position position, List<String> skills,
                   List<String> projects, int subordinatesQuantity) {
        super(name, surname, position, skills);
        this.projects = projects;
        this.subordinatesQuantity = subordinatesQuantity;
    }

    public List<String> getProjects() {
        return projects;
    }

    public void setProjects(List<String> projects) {
        this.projects = projects;
    }

    public int getSubordinatesQuantity() {
        return subordinatesQuantity;
    }

    public void setSubordinatesQuantity(int subordinatesQuantity) {
        this.subordinatesQuantity = subordinatesQuantity;
    }

    public String getProjectsAsString() {
        return projects.stream()
                .reduce((first, second) -> first + ", " + second)
                .get();
    }

    @Override
    void print() {
        System.out.println("Manager: " + getSurname() + " " + getName());
        System.out.println("Position: " + getPosition());
        System.out.println("Skills: " + getSkillsAsString());
        System.out.println("Projects: " + getProjectsAsString());
        System.out.println("Subordinates quantity: " + getSubordinatesQuantity());
    }

}
